### PR TITLE
fix all the warns.

### PR DESCRIPTION
**Pull request by pkofod**
I know I mistyped in the branch name :)

### CODE REVIEW

File: src/optim_tests/multivariate/quad_transforms.jl [hunk 1]
```diff
@@ -126,21 +126,21 @@ end
...
     else
-        retval = F[:Q]'*F[:Q]
+        retval = F.Q'*F.Q
     end
     retval
 end
 
 # TODO: From Julia 0.7 onwards, we can use Base.Test.guardsrand() to restore the existing seed
-oldseed = copy(Base.GLOBAL_RNG) # Store current seed
+oldseed = copy(GLOBAL_RNG) # Store current seed
```
Comment (anriseth) [inline src/optim_tests/multivariate/quad_transforms.jl]: You can use `Test.guardsrand` here now.
Example: https://github.com/JuliaLang/julia/blob/master/test/error.jl#L9

File: src/optim_tests/multivariate/MultivariateProblems.jl [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module MultivariateProblems
 
-import Base.gradient
+import LinearAlgebra: gradient, dot
```
Comment (anriseth) [inline src/optim_tests/multivariate/MultivariateProblems.jl]: I don't think we need to import gradient anymore :)

File: src/optim_tests/multivariate/unconstrained.jl [hunk 3]
```diff
@@ -1,9 +1,13 @@
 module UnconstrainedProblems
 
-using ..OptimizationProblem
+import ..OptimizationProblem
 
 export OptimizationProblem
 
+import LinearAlgebra: Diagonal, qr, dot
+import SparseArrays: sparse
+import Random: srand, GLOBAL_RNG
```
Comment (anriseth) [inline src/optim_tests/multivariate/unconstrained.jl]: `Test.guardsrand` is better than using `Random` here (see comment above)

File: src/optim_tests/multivariate/quad_transforms.jl (hunk 1 above)
Comment (pkofod) [reply src/optim_tests/multivariate/quad_transforms.jl]: Interesting

File: src/optim_tests/multivariate/MultivariateProblems.jl (hunk 2 above)
Comment (pkofod) [reply src/optim_tests/multivariate/MultivariateProblems.jl]: Ah, right! I was just going by the warnings, but that cannot know that we actually didn't want to extend LinearAlgebra's/Base's gradient.

File: src/OptimTestProblems.jl [hunk 4]
```diff
@@ -1,8 +1,11 @@
 module OptimTestProblems
 
+import LinearAlgebra: gradient
```
Comment (anriseth) [inline src/OptimTestProblems.jl]: Another "unnecessary" gradient ;)

Comment (pkofod) [reply src/OptimTestProblems.jl]: didn't you have a thesis to submit???

Comment (anriseth) [reply src/OptimTestProblems.jl]: Haha, yes I'm gonna leave it to you now :laughing:

Comment (pkofod) [reply src/OptimTestProblems.jl]: congrats btw 🎆 (is it submitted?)

Comment (anriseth) [reply src/OptimTestProblems.jl]: I need to temporarily turn off GitHub emails :p

Comment (anriseth) [reply src/OptimTestProblems.jl]: In a few hours; UK academia is still fast asleep :see_no_evil:

**Pull request merged**